python/pyarrow/types.pxi

```diff
@@ -2110,11 +2110,28 @@ def float64():
 
 cpdef DataType decimal128(int precision, int scale=0):
     """
-    Create decimal type with precision and scale and 128bit width.
+    Create decimal type with precision and scale and 128-bit width.
+
+    Arrow decimals are fixed-point decimal numbers encoded as a scaled
+    integer.  The precision is the number of significant digits that the
+    decimal type can represent; the scale is the number of digits after
+    the decimal point (note the scale can be negative).
+
+    As an example, ``decimal128(7, 3)`` can exactly represent the numbers
+    1234.567 and -1234.567 (encoded internally as the 128-bit integers
+    1234567 and -1234567, respectively), but neither 12345.67 nor 123.4567.
+
+    ``decimal128(5, -3)`` can exactly represent the number 12345000
+    (encoded internally as the 128-bit integer 12345), but neither
+    123456000 nor 12345600.
```

Comment:
It also would be unable to represent the numbers 123450000 and 1234500 correct? Would it be clearer to use those similar to the decimal example we use above in type.h? Where the sig figs are exactly the same, but the decimal is the only thing that is different?